source/troubleshooting/connection-errors.txt

```diff
@@ -0,0 +1,311 @@
+         verify that you have specified a valid :setting:`net.bindIp`
+         setting in your server :manual:`configuration file
+         </reference/configuration-options/>`.
+
+         Hostname and Port
+         `````````````````
+ 
+         If your self-hosted server is listening on an external network
+         via :setting:`net.bindIp`, verify that you have specified your
+         :guilabel:`Hostname` and :guilabel:`Port` correctly in the
+         |compass-short| connect dialog.
+ 
+         MongoDB Server Moved or Shutdown
+         ````````````````````````````````
+
+         .. include:: /includes/fact-server-moved-or-shutdown.rst
+ 
+         Firewall
+         ````````
+ 
+         If you believe your firewall may be actively blocking
+         connections to the port(s) your database(s) are operating on,
+         try connecting from a different location, such as a home or
+         internet cafe network. You can also visit http://portquiz.net/
+         to test your connection to different ports.
+
+     - id: atlas
+       name: Atlas Cluster
+       content: |
+
+         If you are connecting to an |service| cluster, first follow
+         the procedure described in the :atlas:`Connect via Compass
+         </compass-connection/>` page in the |service| documentation.
+         The steps on that page will help ensure that your connection
+         settings are optimized for |service|.
+
+         MongoDB Server Moved or Shutdown
+         ````````````````````````````````
+ 
+         .. include:: /includes/fact-server-moved-or-shutdown.rst
+
+         Firewall
+         ````````
+ 
+         If you believe your firewall may be actively blocking
+         connections to your database, try connecting from a different
+         location, such as a home or internet cafe network.
+
+         |service| clusters operate on port 27017. You must be able to
+         access this port to access your cluster. To check if your
+         local network blocks connections to port 27017, visit
+         http://portquiz.net:27017/.
+
+         Atlas IP Whitelist Does Not Include Your Address
+         ````````````````````````````````````````````````
+
+         |service| only allows connections to a cluster from addresses
+         listed in the project :atlas:`IP whitelist
+         </security-whitelist/>`. Ensure that you have
+         whitelisted your IP address so you can connect to your
+         cluster.
+
+Insert Document Button Disabled and Cannot Modify Documents
+-----------------------------------------------------------
+
+When using |compass-short|, you may be in a state where you cannot
+perform any write operations, and can only read data.
+
+Possible Observations
+~~~~~~~~~~~~~~~~~~~~~
+
+When you are in this state:
+
+- The :guilabel:`Create Collection` and :guilabel:`Insert Document`
+  buttons are disabled.
+
+- You cannot modify any of your documents.
+
+Likely Cause
```

Comment:
I know this is silly, but what if people don't know they downloaded the Readonly version of Compass? Should we say anything about that and possible link back to the landing page so that they can learn a bit more about the different editions of Compass?

Reply:
Looks great